embark.el

```diff
@@ -466,55 +471,63 @@ There are three kinds:
 
 (defun embark-target-active-region ()
   "Target the region if active."
+  ;; TODO consider returning a string
   (when (use-region-p) '(region . <region>)))
 
 (autoload 'dired-get-filename "dired")
 
 (defun embark-target-file-at-point ()
   "Target file at point.
-This function mostly relies on `ffap-file-at-point', with two exceptions:
-
-1. In `dired-mode', it uses `dired-get-filename' instead.
-
-2. In `emacs-lisp-mode', it only calls `ffap-file-at-point' if
-   point is in a string or comment, or if it is on symbol
-   preceded by `require' or `use-package'."
-  (when-let ((file (cond
-                    ((derived-mode-p 'dired-mode)
-                     (dired-get-filename t 'no-error-if-not-filep))
-                    ((derived-mode-p 'emacs-lisp-mode)
-                     (when (or (nth 3 (syntax-ppss)) ; in a string
-                               (nth 4 (syntax-ppss)) ; or comment
-                               (save-excursion
-                                 (unless (looking-at "\\_<")
-                                   (forward-symbol -1))
-                                 (forward-symbol -1)
-                                 (looking-at "use-package\\|require")))
-                       (ffap-file-at-point)))
-                     (t (ffap-file-at-point)))))
-    (cons 'file (abbreviate-file-name file))))
+This function mostly relies on `ffap-file-at-point', with one exception:
+In `dired-mode', it uses `dired-get-filename' instead."
+  (if-let (file (and (derived-mode-p 'dired-mode)
+                     (dired-get-filename t 'no-error-if-not-filep)))
+      (save-excursion
+        (end-of-line)
+        `(file ,(abbreviate-file-name file)
+               ,(save-excursion
+                  (re-search-backward " " (line-beginning-position) 'noerror)
+                  (1+ (point)))
+               . ,(point)))
+    (when-let (file (ffap-file-at-point))
+      `(file ,(abbreviate-file-name file)
+             . ,(bounds-of-thing-at-point 'filename)))))
```

Comment:
Does use the bounds from thingatpt really work? I used ffap because it's much smarter and detects file names in many more cases. For example in `(require 'lib)` ffap knows `lib` is a file. Does thingatpt return the correct bounds in that case?

Reply:
Not really, there should probably be a TODO here to make this more general. It is a best effort thing, we may get wrong bounds but the only thing which will break is the hl. Unfortunately ffap does not make the bounds available.